### PR TITLE
fix: fixed error handling in persister

### PR DESCRIPTION
**Pull request by mitar**
## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](docs/docs).

### CODE REVIEW

**Comment by mitar** [comment]
I also noticed duplicated code, so I removed that.

File: persistence/sql/persister_oauth2.go [hunk 1]
```diff
@@ -391,50 +393,23 @@ func (p *Persister) FlushInactiveAccessTokens(ctx context.Context, notAfter time
 
 		if i != j {
 			err = p.Connection(ctx).RawQuery(
-				fmt.Sprintf("DELETE FROM %s WHERE signature in (?)", OAuth2RequestSQL{Table: sqlTableAccess}.TableName()),
+				fmt.Sprintf("DELETE FROM %s WHERE signature in (?)", OAuth2RequestSQL{Table: table}.TableName()),
 				signatures[i:j],
 			).Exec()
+			if err != nil {
+				break
```
Comment (aeneasr) [inline persistence/sql/persister_oauth2.go:400]:
```suggestion
				return sqlcon.HandleError(err)
```

**Review by aeneasr — COMMENTED** [review]
Thank you! Looking good, just a minor change

File: persistence/sql/persister_oauth2.go (hunk 1 above)
Comment (mitar) [reply persistence/sql/persister_oauth2.go]: Why this change? You already have `return sqlcon.HandleError(err)` at the end of the function which handles this?

Comment (aeneasr) [reply persistence/sql/persister_oauth2.go]: Because it's explicit

**Comment by aeneasr** [comment]
🙏

File: persistence/sql/persister_oauth2.go (hunk 1 above)
Comment (mitar) [reply persistence/sql/persister_oauth2.go]: Then you can probably just make the last line in the function `return nil`?

**Pull request merged**